examples/user_guide/Reactive_Expressions.ipynb

```diff
@@ -0,0 +1,258 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "3ecf9ffc-96a0-4457-bc89-43cac0099e01",
+   "metadata": {},
+   "source": [
+    "## Reactive Functions & Expressions\n",
+    "\n",
+    "In the [Dependencies and Watchers](Dependencies_and_Watchers.ipynb) guide we discovered how to express dependencies and write callbacks that are invoked when parameter values change. This imperative style of expressing dynamic behavior is powerful but can also end up resulting in complex code that is hard to read and maintain. Therefore Param 2.0 introduces a set of primitives that make it possible to express complex dependencies in a declarative form resulting in reactive expressions that are re-evaluated automatically when required.\n",
+    "\n",
+    "The reactive programming model is one you might be familiar with from Excel, where formulas can reference cells or ranges and dynamically (or rather reactively) recompute when the inputs to a formula changes. In Param, Parameter objects represent the inputs or references in a formula and reactive expressions replace the concept of a formula. Additionaly `param.bind` allows the creation of a reactive function with arbitrary inputs."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cbfdfa4b-a04e-465b-bf1f-2910cccd870f",
+   "metadata": {},
+   "source": [
+    "## Reactive Functions\n",
+    "\n",
+    "Let us start by creating a `Parameterized` class with some parameters, here named `a` and `b`. Additionally we import `param.ipython` to allow reactive components to display correctly and re-render themselves when the inputs change."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f6bf6371-8dc8-4247-afaa-3ce872ed3371",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import param\n",
+    "import param.ipython\n",
+    "\n",
+    "class Parameters(param.Parameterized):\n",
+    "    \n",
+    "    a = param.Number(default=1)\n",
+    "\n",
+    "    b = param.Number(default=1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1c4434e7-9b5d-4a5b-b9de-c444dae42e36",
+   "metadata": {},
+   "source": [
+    "Now we will create a reactive function using `param.bind`. We start by declaring a function that adds two parameters and then bind parameters `a` and `b` to that function:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c0a4c64e-ee64-405d-bace-605551b34234",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def add(a, b):\n",
+    "    return a+b\n",
+    "\n",
+    "p = Parameters()\n",
+    "\n",
+    "reactive_add = param.bind(add, p.param.a, p.param.b)\n",
+    "\n",
+    "reactive_add"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9094768c-5a39-448e-9721-e6253a2a55cb",
+   "metadata": {},
+   "source": [
+    "You can see that this reactive function renders the output and when we update one of the inputs to the function the output will reflect the updated result."
```

Comment:
The `IPython` display handlers hooks into the Jupyter display machinery, so anything that renders in Jupyter will render correctly. This will certainly differ from Panel in some respects. Note however that `pn.extension` will override the default display handler, so there's no mixing of display handlers. Either you use the `param.ipython` display handler or you use the `panel` display handler.